src/components/LocationTile/LocationEdit.jsx

```diff
@@ -26,29 +31,44 @@ function LocationEdit({ locationId, setLocations, showEdit, setShowEdit }) {
   // Handle form submission
   const handleSubmit = async (e) => {
     e.preventDefault();
-    const updatedLocation = await updateLocation(locationId, locationToUpdate); 
+    setIsLoading(true); // Disable form while submitting
+    try {
+      const updatedLocation = await updateLocation(locationId, locationToUpdate);
 
-    // Update the location in the array
-    setLocations((prevLocations) =>
-      prevLocations.map((location) =>
-        location.id === updatedLocation.id ? updatedLocation : location
-      )
-    );
-
-    setShowEdit(false); // Hide the form after submission
-    setLocationToUpdate({ name: "", address: "" }); // Reset the form inputs
+      if (updatedLocation) {
+        // Update the location in the array
+        setLocations((prevLocations) =>
+          prevLocations.map((location) =>
+            location.id === updatedLocation.id ? updatedLocation : location
+          )
+        );
+        setShowEdit(false); // Hide the form after submission
+      }
+    } catch (error) {
+      console.error("Failed to update location:", error);
+    } finally {
+      setIsLoading(false); // Re-enable form
+    }
   };
 
+  // Handle deletion
   const handleDelete = async () => {
-    await deleteLocation(locationId);
+    setIsLoading(true); // Disable buttons while deleting
+    try {
+      await deleteLocation(locationId);
 
-    // **Remove the deleted location from the state**
-    setLocations((prevLocations) =>
-      prevLocations.filter((location) => location.id !== locationId)
-    );
+      // Remove the deleted location from the state
+      setLocations((prevLocations) =>
+        prevLocations.filter((location) => location.id !== locationId)
+      );
 
-    // **Hide the form after deletion**
-    setShowEdit(false);
+      // Redirect to profile page after deletion
+      navigate("/profile");
```

Comment:
Curious why redirect back to /profile instead of setShowEdit, might cause an unnecessary network request